File: Javascript/chartmeleon.js
function Chartmeleon() {
  this.getShades = function(hue, numberOfShades) {
    var lightness = 0;
    var saturation = 0;
    var lightnessSteps = numberOfShades === 1 ? 50 : 90 / numberOfShades;
    var saturationStepsDistance = 100 / Math.ceil(numberOfShades / 2);
    var shades = new Array();

    for (var i = 1; i <= numberOfShades; i++) {
      lightness += lightnessSteps;
      saturation += saturationStepsDistance;
      shades.push("hsl(" + hue + ", " + saturation + "%, " + lightness + "%)");
    }

    return shades;
  };

  this.getColorSet = function(numberOfHues, numberOfShades, random) {
    var hueStepDistance = 360 / numberOfHues;
    var hue = random === true ? Math.floor(Math.random() * 361) : 0;
    var colors = new Array();

    for (var i = 0; i < numberOfHues; i++) {
      colors.push(this.getShades(hue, numberOfShades));
      hue += hueStepDistance;
    }

    return colors;
  };
}

File: Java/src/main/java/com/chartmeleon/Chartmeleon.java
package com.chartmeleon;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class Chartmeleon {

	public static List<String> getShades(int hue, int numberOfShades) {
		double lightness = 0;
		double saturation = 0;
		double lightnessSteps = numberOfShades == 1 ? 50 : 90 / numberOfShades;
		double saturationStepsDistance = 100 / Math.ceil(numberOfShades / 2.0);
		List<String> shades = new ArrayList<String>();

		for (int i = 0; i < numberOfShades; i++) {
			lightness += lightnessSteps;
			saturation += saturationStepsDistance;
			shades.add("hsl(" + hue + ", " + saturation + "%, " + lightness + "%)");
		}

		return shades;
	}

	public static List<List<String>> getColorSet(int numberOfHues, int numberOfShades, boolean random) {
		int hueStepDistance = 360 / numberOfHues;
		int hue = random == true ? new Random().ints(0, (360 + 1)).findFirst().getAsInt() : 0;
		List<List<String>> colors = new ArrayList<List<String>>();

		for (int i = 0; i < numberOfHues; i++) {
			colors.add(getShades(hue, numberOfShades));
			hue += hueStepDistance;
		}

		return colors;
	}
}